src/staticdata.c

```diff
@@ -922,14 +937,17 @@ static void jl_insert_into_serialization_queue(jl_serializer_state *s, jl_value_
     else if (jl_typetagis(v, jl_module_tag << 4)) {
         jl_queue_module_for_serialization(s, (jl_module_t*)v);
     }
+    else if (jl_is_binding_partition(v)) {
+        jl_binding_partition_t *bpart = (jl_binding_partition_t*)v;
+        jl_queue_for_serialization_(s, decode_restriction_value(jl_atomic_load_relaxed(&bpart->restriction)), 1, immediate);
+        jl_queue_for_serialization_(s, get_replaceable_field((jl_value_t**)&bpart->next, 0), 1, immediate);
```

Comment:
IIUC, this will not serialize/deserialize correctly if the user (or inference) creates a reference to it. We need to determine how to correctly reattach them to the Binding / module that owns them (maybe by recording the location of all BindingPartitions in the system, as a pre-processing pass, so we can query here the Binding that owns it?)

Reply:
What's the particular issue you're imagining? Currently they're attached 1:1. In the full thing, you're not allowed to create new partitions for a binding already in the image, so they don't need to be ported.

Reply:
Ah right, the code talks about "walking" the list of them. But it is currently always guaranteed to actually just be a single value that gets mutated in place (e.g. when a `using` finally gets resolved)?

Reply:
yes